mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/task/CassandraRecomputeCurrentQuotasServiceTest.java

```diff
@@ -107,4 +117,46 @@ public UserQuotaRootResolver userQuotaRootResolver() {
     public RecomputeCurrentQuotasService testee() {
         return testee;
     }
+
+    @Disabled("JAMES-3138 Doens't make the reset fail...")
+    @Test
+    void recomputeCurrentQuotasShouldReturnPartialWhenFailureAtReset() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        MailboxSession session = sessionProvider().createSystemSession(USER_1);
+        mailboxManager().createMailbox(MAILBOX_PATH, session);
+
+        MessageManager messageManager = mailboxManager().getMailbox(MAILBOX_PATH, session);
+        appendAMessageForUser(messageManager, session);
+
+        cassandraCluster.getCassandraCluster().getConf().registerScenario(fail()
+            .times(1)
+            .whenQueryStartsWith("UPDATE currentQuota SET"));
+
+        assertThat(testee().recomputeCurrentQuotas(new RecomputeCurrentQuotasService.Context()))
+            .isEqualTo(Task.Result.PARTIAL);
+    }
+
+    @Disabled("JAMES-3138 Doens't make the reset fail...")
+    @Test
+    void recomputeCurrentQuotasShouldUpdateContextWhenFailureAtReset() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        MailboxSession session = sessionProvider().createSystemSession(USER_1);
+        mailboxManager().createMailbox(MAILBOX_PATH, session);
+
+        MessageManager messageManager = mailboxManager().getMailbox(MAILBOX_PATH, session);
+        appendAMessageForUser(messageManager, session);
+
+        cassandraCluster.getCassandraCluster().getConf().registerScenario(fail()
+            .times(1)
+            .whenQueryStartsWith("UPDATE currentQuota SET"));
```

Comment:
I really thought that should be the way to make it fail. Well it doesn't, the operation is still a success at the end. Really not sure where I'm wrong here...

Reply:
Your issue is that you are not using the CassandraAsyncExacutor within the currentQuotaDAO.

Either:
 - We support failure injection for synchronous queries within "TestingSession"
 - Or we convert the DAO to use CassandraAsyncExecutor.

Or... Both.

Reply:
I wanted to try to reactify all of this in an other PR... I guess I can still make the Dao async without the rest though, would be a first step. Thanks !

Reply:
Done, working now. Thanks !